### DISCUSSION TITLE
RP2040 wrong firmware

### DISCUSSION THREAD

**Discussion by onenoter** [ZuluSCSI/ZuluSCSI-firmware#348] [Q&A] [answered]
Hi guys,
I had my SD card connected to my Amiga and despite having create-hardfiles on it, it went into raw-mode. I did not realize that I could crash my SD this way, so I Amiga-formatted it. It worked fine. But after a while, I missed the hard-files so I decided to reformato it in my pc - it was unreadable due to the Amiga format. This made the card re-readable in the pc, but the RP2040 could not read it anymore.

So I decided to flash my RP2040 to the newest firmware with another card, hoping it would solve my problem. But I clicked on the wrong firmware and installed the BS2-version. The firmware installed (one second fast blinking) and then, none of my SD-cards were recognized anymore.
I tried to reflash it via the USB-Drive in bootloader-mode (just placing the newest or the former newest firmware for the 2040 on it - it vanishes) without any success. My serial output is still:
[10ms] Platform: ZuluSCSI BS2
[10ms] FW Version: 23.11.27-release Nov 28 2023 01:25:39
[11ms] SCSI termination is handled by a hardware jumper
[11ms] Flash chip size: 4096 kB
[12ms] SCSI target/disk mode, acting as a SCSI disk
[36ms] SDIO SD card error on line 445, error code 2
[39ms] SDIO SD card error on line 445, error code 2
[42ms] SDIO SD card error on line 445, error code 2
[45ms] SDIO SD card error on line 445, error code 2
[45ms] SD card init failed, sdErrorCode: 2 sdErrorData: 0

Any suggestions?

**Comment by aperezbios** [comment, marked as answer]
@onenoter your only option here is use the only recovery mechanism that's available, which is the built-in ROM bootloader, which will allow you to copy a .uf2 file to the device. It sounds like you attempted this already, but something with the process did not happen correctly, so please try again. You _need_ to use https://github.com/ZuluSCSI/ZuluSCSI-firmware/releases/download/v2023.11.27/ZuluSCSI_RP2040_2023-11-27.uf2 . The .bin file will not work, as the RP2040 bootloader does not understand the format. Plese let us know how it goes.

**Comment by onenoter** [comment]
Thank you, fast and competent :) ! I used the .bin-file which did not work. The .uf2 worked fine, all back to normal!

The device worked flawlessly in raw-mode before, I am looking forward to use it again and in file-mode. The manual seems to be a little rudimental and maybe you should integrate a model-checking in the update-procedure (for stupid users like me), but the raw SD-card was recognized by the Amiga like a perfect harddrive. My real harddrives (U320 server-models) are not so easy recognized. Good work :) .

Since this is now in "discussion", maybe I can suggest some manual-improvements here:
- Copy the information how to create the hardfiles from the firmware-page to the part you mention them (Quickstart Guide)
- Mention the way to let the Pi create the files at the same position, add the flashing LED and that the creation of an 1GB-file does take about 1 minute.
- Correct the spelling mistake: "ZuluSCSI firmware can also create image files itself. To do this, create a text file with filename such as Create 1024M HD40.txt." It should be a HD4 instead of HD40, right? The number does represent the SCSI-id, right? If so, please add an explaination for the number.
- Add a "look on the right column for new firmware and copy the proper .bin-file" to the card" in the firmware section. If you are not used to github, you will search for hours for the real download location.
- Mention the SD-format software from the SD-initiative.
- Create a better ini-manual that explains in detail how to create custom drives etc.
- Mention that the bootloader-directory can only read the uf2-files and not the bins.

Just trying to be helpful and give back something. I know it is nearly impossible to write a manual suited for every stupid user :) .
